### PR TITLE
Add Raw Byte Entity Serialization (1.15)

### PR DESCRIPTION
**Pull request by Proximyst**
Adds an unsafe API (feel free to add a "safe" one like last time, Aikar) to serialize & deserialize entity data.

### CODE REVIEW

**Comment by aikar** [comment]
I have a patch for this in EmpireCraft but it wasnt ready for Paper yet due to how it was impl. def on board with it being inPaper, but want to make sure its in same format as mine so I can move to it cleanly.

https://github.com/starlis/empirecraft/blob/master/patches/server/0014-Entity-Serialization.patch

If you can compare

**Comment by aikar** [comment]
first glance looks ok, dont need to be same methods as me, and im ok with the entity type addition and we can just handle null potential if we ever used it ourselves (but prob wont, but i can also just patch my fork)

the spawn api is a big one to add here.

**Comment by Proximyst** [comment]
The API now uses the EMC implementation and has extracted the ability to serialize & deserialize NBT compound tags into their own methods. This should result in less code duplication and less misreading for the future.

File: Spigot-API-Patches/0214-Add-Raw-Byte-Entity-Serialization.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index d3ec5084e33dff038d54cdd2aeb703a3eb25f7a7..2d62f5c3c1d50c3218707d2fec35965b0500cdfa 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -2,6 +2,8 @@ package org.bukkit;
+ 
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
++import org.bukkit.entity.Entity; // Paper
++import org.bukkit.entity.EntityType; // Paper
```
Comment (Proximyst) [inline Spigot-API-Patches/0214-Add-Raw-Byte-Entity-Serialization.patch]: missed this one :( Will remove

**Comment by Proximyst** [comment]
As 1.15.2 is moving towards EOL and this has little probability of ever being merged into it, I'll close the PR. There's still the 1.16 variant, though: #3628.

**Pull request closed without merging**